src/Verify.MSTest.SourceGenerator/Emitter.cs

```diff
@@ -59,11 +59,13 @@ void WriteParentTypes(ClassToGenerate classToGenerate)
             .AppendLine("{")
                 .IncreaseIndent()
                 .AppendLine(GeneratedCodeAttribute)
-                .AppendLine("public TestContext TestContext")
+                .AppendLine(classToGenerate.HasExistingTestContextProperty
```

Comment:
I'm not sure if I agree with this change. What's the scenario where it's needed?

`new` _hides_ the base member, but it doesn't override it. That means code that accesses the base class as the base type will continue to use the "hidden" accessor. Consider a scenario like this where several test classes share a base class that defines a set of "helpers":

```csharp
public abstract class Base
{
    public TestContext TestContext { get; set; } = null!;

    protected void Helper() => if (TestContext is null) { throw new("TestContext is null"); }
}

[TestClass]
public class Derived : Base
{
    // This line simulates the generator, where it's adding the TestContext property with the `new` keyword
    public new TestContext TestContext { get; set; } = null!;

    [TestMethod]
    public void TestMethod1()
    {
        Helper(); // throws!
        Assert.IsTrue(true);
    }
}
```

This code throws in `Helper()` because MSTest won't do property injection on the base class (correctly because it isn't a `[TestClass]` itself), and Base can't see the TestContext in `Derived` because the base property isn't marked `virtual`.

If there are cases where the generator would be hiding a base member, I'd prefer compilation errors to using `new` and introducing subtle bugs. However, I thought `InheritanceTests` covered all the permutations. Are there scenarios where the generator is hiding the base member?

Reply:
[This](https://github.com/VerifyTests/Verify/blob/main/src/Verify.MSTest.SourceGenerator.Tests/InheritanceTests.cs#L62-L79) test with an existing `TestContext` in the base class is the test that needed the new.

We just need a way to solve that problem.

Reply:
Ah, well your diagnostics assertions are already paying off then :)

I think this is a case of under specification in the tests on my part. This is the same as the case above, where the user's code is fundamentally ambiguous and needs to be changed.

Thus, I think the previous generated code was correct in the sense that we shouldn't use `new` and causing a CS0108 is the right thing to do in this case.

How does this sound for a plan? We should revert the `HasExistingTestContextProperty` and associated code, and instead update the test so that rather than asserting that there are no diagnostics, assert that for this test CS0108 _should_ exist.

Does that make sense? Let me know if you have any questions!

Reply:
Ok, I've left the generator as is, and updated the test to expect the diagnostic error.